provider/uprops/src/binary.rs

```diff
@@ -0,0 +1,97 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::upropdump_serde;
+use icu_provider::prelude::*;
+use icu_provider::iter::IterableDataProviderCore;
+use icu_uniset::provider::*;
+use icu_uniset::UnicodeSetBuilder;
+use std::borrow::Cow;
+use std::fs;
+use std::path::PathBuf;
+
+pub struct BinaryPropertiesDataProvider {
+    root_dir: PathBuf,
+}
+
+impl BinaryPropertiesDataProvider {
+    pub fn new(root_dir: PathBuf) -> Self {
+        BinaryPropertiesDataProvider { root_dir }
+    }
+    fn get_toml_as_string(&self, name: &str) -> Result<String, DataError> {
+        let mut path: PathBuf = self.root_dir.clone().join(name);
+        path.set_extension("toml");
+        fs::read_to_string(&path).map_err(DataError::new_resc_error)
```

Comment:
Nit: It's preferable if you attach the filename to the error.  Most other ICU4X components do this by having an `Error` specific to their crate that has an `Io` variant.